src/diagnostics/land_compute_methods.jl

```diff
@@ -0,0 +1,146 @@
+"""
+    @land_compute name model compute
+
+
+Macro generating a function to compute a land diagnostic,
+needed in the ClimaDiagnostics framework.
+
+For example,
+@land_compute "soil_net_radiation" SoilCanopyModel p.soil.R_n
+
+generates the function
+
+function compute_soil_net_radiation!(out, Y, p, t, land_model::SoilCanopyModel)
+    if isnothing(out)
+        return copy(p.soil.R_n)
+    else
+        out .= p.soil.R_n
+    end
+end
+"""
+macro land_compute(name, model, compute)
+    functions = Expr[]
+    function_name = Symbol("compute_", name, "!")
+    function_body = esc(quote
+        function $function_name(out, Y, p, t, land_model::$model)
+            if isnothing(out)
+                return copy($compute)
+            else
+                out .= $compute
+            end
+        end
+    end)
+    push!(functions, function_body)
+    return quote
+        $(functions...)
+    end
+end
+
+### BucketModel ###
+
+# variables stored in p (diagnostics variables stored in the cache)
+@land_compute "aerodynamic_resistance" BucketModel p.bucket.turbulent_fluxes.r_ae
+@land_compute "albedo" BucketModel p.bucket.α_sfc
+@land_compute "latent_heat_flux" BucketModel p.bucket.turbulent_fluxes.lhf
+@land_compute "net_radiation" BucketModel p.bucket.R_n
+@land_compute "sensible_heat_flux" BucketModel p.bucket.turbulent_fluxes.shf
+@land_compute "surface_air_density" BucketModel p.bucket.ρ_sfc
+@land_compute "surface_specific_humidity" BucketModel p.bucket.q_sfc
+@land_compute "surface_temperature" BucketModel p.bucket.T_sfc
+@land_compute "vapor_flux" BucketModel p.bucket.turbulent_fluxes.vapor_flux
+
+# variables stored in Y (prognostic or state variables)
+@land_compute "snow_water_equivalent" BucketModel Y.bucket.σS
+@land_compute "soil_temperature" BucketModel Y.bucket.T
+@land_compute "subsurface_water_storage" BucketModel Y.bucket.W
+@land_compute "surface_water_content" BucketModel Y.bucket.Ws
+
+### SoilCanopyModel ###
+
+# variables stored in p (diagnostics variables stored in the cache)
+
+## Canopy Module ##
+
+# Canopy - Solar Induced Fluorescence
+@land_compute "solar_induced_fluorescence" SoilCanopyModel p.canopy.sif.SIF
+
+# Canopy - Autotrophic respiration
+@land_compute "autotrophic_respiration" SoilCanopyModel p.canopy.autotrophic_respiration.Ra
+
+# Canopy - Conductance
+@land_compute "medlyn_term" SoilCanopyModel p.canopy.conductance.medlyn_term
```

Comment:
Im not sure we need this one, we care more about gs and transpiration

Reply:
I agree... I will remove it for now